### ISSUE TITLE
Athena select query, with order by, returns unordered dataframe.

### ISSUE THREAD

**Issue by fschiavinato-lydian** [awslabs/aws-data-wrangler#785]
Hi, I am using wrangler to read queries from athena and I found that the order by clause is not working correctly. 
I am using the method awswrangler.athena.read_sql_query. 
I notice that if I set ctas_approach=False it works correctly but it's slower.
I am using awswrangler==2.7.0.

**Comment by kukushking** [comment]
Thanks for reporting this @fschiavinato-lydian - indeed, at the moment with `ctas_approach=True` the order of the results isn't guaranteed because the CTAS table may consist of multiple Parquet files that are read directly from S3 in no particular order. In fact, Parquet & reading multiple files in parallel is exactly what provides performance benefits you're seeing with CTAS approach.

One thing we may consider adding is CTAS bucketing. It would then become possible to set the number of files to 1 that will solve the ordering problem and will keep most of the performance benefits of CTAS.

**Comment by kukushking** [comment]
@fschiavinato-lydian I added CTAS bucketing - below an example how you can force `read_sql_query` to output exactly one result file reading which you shouldn't have any ordering problems:
```
df = wr.athena.read_sql_query(
        sql=f"SELECT * FROM {glue_table} ORDER BY c1",
        ctas_approach=True,
        database=db,
        ctas_database_name=db_ctas,
        ctas_temp_table_name=tbl,
        ctas_bucketing_info=(["c0"], 1),
        s3_output=path,
    )
```
Of course this should be used with caution on big datasets. 

The functionality will be included in 2.10 release but you can try it out right now by installing from source:
```
pip uninstall awswrangler -y
pip install git+https://github.com/awslabs/aws-data-wrangler.git@main
```

**Comment by fifilura** [comment]
@kukushking Unfortunately the approach to force one parquet file is not enough.

I have verified this myself. Only one file is created using the bucketing trick, but the result is still not ordered. For me it happens around 20% of the time when querying a table of 100000 rows.

(It is a bit convoluted to make a testcase of it since the data is internal.)

The claim is also supported by this documentation https://docs.aws.amazon.com/athena/latest/ug/ctas-considerations-limitations.html#ctas-considerations-limitations-order-by-ignored

I think at least it should be brought up in the pros and cons here: https://aws-sdk-pandas.readthedocs.io/en/stable/stubs/awswrangler.athena.read_sql_query.html
